src/lcgs/parse.rs

```diff
@@ -1009,4 +1041,18 @@ mod tests {
         println!("{:?}", parser.parse(input));
         assert!(parser.parse(input).is_ok());
     }
+
+    #[test]
+    fn test_reserved_keyword_01() {
+        let input = br"label legal_ident = 1;";
+        let parser = identifier();
+        assert!(parser.parse(input).is_err());
+    }
+
+    #[test]
+    fn test_reserved_keyword_02() {
+        let input = br"label label = 1;";
+        let parser = identifier();
+        assert!(parser.parse(input).is_err());
+    }
```

Comment:
It would be nice but I don't think it's necessary. Since the parser is static, it cannot be easily written, as I tried in the following:
```rust
    fn test_reserved_keyword_02() {
        for RESERVED_KEYWORD in RESERVED_KEYWORDS.iter() {
            let input = format!("label {} = 1;", RESERVED_KEYWORD);
            let parser = identifier();
            assert!(parser.parse(input.clone().as_bytes()).is_err());
        }
    }
```

Let's just ignore that for now.

Reply:
I am going to have to fix the `'static` lifetime to parse from the CLI too, so I am looking into it.

Reply:
why not hand write a test for every keyword?

Reply:
I figured out a bug I made, and have now written a test for all keywords 6f47d3e